### PR TITLE
feat: Mouse scroll

### PR DESCRIPTION
**Pull request by jeevithakannan2**
## Type of Change
- [x] New feature
- [x] UI/UX improvement

## Description
- Add mouse scroll in all directions.

https://github.com/user-attachments/assets/d666c77f-37ec-4469-8ff7-130d14da4e23

## Testing
- Tested on arch linux with no issues.

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: tui/src/main.rs [hunk 1]
```diff
@@ -70,15 +72,23 @@ fn run(
...
-                return Ok(());
+                if !state.handle_key(&key) {
+                    return Ok(());
+                }
+            }
+            Event::Mouse(mouse_event) => {
+                if !state.handle_mouse(&mouse_event) {
+                    return Ok(());
+                }
+                continue;
```
Comment (lj3954) [inline tui/src/main.rs:89]:
```suggestion
```
No need for this.

File: tui/src/confirmation.rs [hunk 2]
```diff
@@ -84,6 +84,21 @@ impl FloatContent for ConfirmPrompt {
...
+            MouseEventKind::ScrollDown => {
+                self.scroll_down();
+                ConfirmStatus::None
+            }
+            MouseEventKind::ScrollUp => {
+                self.scroll_up();
+                ConfirmStatus::None
+            }
+            _ => ConfirmStatus::None,
+        };
```
Comment (lj3954) [inline tui/src/confirmation.rs:98]:
```suggestion
        match event.kind {
            MouseEventKind::ScrollDown => {
                self.scroll_down();
            }
            MouseEventKind::ScrollUp => {
                self.scroll_up();
            }
            _ => {}
        }
```
Why is this match statement returning an unused ConfirmStatus?

File: tui/src/state.rs [hunk 3]
```diff
@@ -416,6 +416,45 @@ impl AppState {
...
+            Focus::FloatingWindow(float) => {
+                float.content.handle_mouse_event(event);
+            }
+            Focus::ConfirmationPrompt(confirm) => {
+                confirm.content.handle_mouse_event(event);
+            }
+            _ => {}
+        }
+        true
+    }
```
Comment (lj3954) [inline tui/src/state.rs:456]: We should consider taking the position of the mouse into account. For example, mouse events outside a floating window should be ignored, and the tablist/list focus should switch if a mouse event occurs in the other.

**Review by lj3954 — CHANGES_REQUESTED** [review]

File: tui/src/confirmation.rs (hunk 2 above)
Comment (jeevithakannan2) [reply tui/src/confirmation.rs:98]: I was wondering the same with handle_key_event

File: tui/src/state.rs (hunk 3 above)
Comment (jeevithakannan2) [reply tui/src/state.rs:456]: Yeah I get it. That was my idea but it requires little bit more time. I have my semesters from next week will need to do that after my semesters. For now this is okayish

Comment (jeevithakannan2) [reply tui/src/state.rs]: Implemented !!

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]
this should be togglable and off by default

**Comment by jeevithakannan2** [comment]
> this should be togglable and off by default

Why would someone want that?

**Comment by nnyyxxxx** [comment]
> > this should be togglable and off by default
> 
> Why would someone want that?

this is a tui application ran in the terminal having mouse interaction without a toggle is poor ux by default

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**